assignment_2_Q1.py

```diff
@@ -0,0 +1,72 @@
+class Node:
+    def __init__(self, item, left = None, right = None):
+        self.item = item
+        self.left = left
+        self.right = right
+
+class BST:
+    def __init__(self, lst = None):
+        self.root = None
+        if lst != None:
+            for item in lst:
+                self.add(item)
+
+    def add(self, item):
+        if self.root == None:
+            self.root = Node(item, None, None)
+        else:
+            # Find where to put the item
+            child_tree = self.root
+            while child_tree != None:
+                parent = child_tree
+                if item < child_tree.item: 
+                    child_tree = child_tree.left 
+                else:
+                    child_tree = child_tree.right
+
+            # child_tree should be pointing to the new node, but we've gone too far
+            # we need to modify the parent nodes
+            if item < parent.item:
+                parent.left = Node(item, None, None)
+            else:
+                parent.right = Node(item, None, None)
+
+
+
+# find all ancestors for a Binary Tree
+def rec_find_ancestors(ptr, key, path):

```

Comment:
What does "rec" stand for? Could you please use a more obvious term for the slow readers like me, or explain in a comment?

Also, can you write a short docstring that explains what the parameters are for?